service/history/ndc/history_replicator.go

```diff
@@ -263,6 +263,23 @@ func (r *HistoryReplicatorImpl) ApplyWorkflowState(
 		return err
 	}
 
+	historyBranchResp, err := r.shard.GetExecutionManager().ParseHistoryBranchInfo(ctx,
+		&persistence.ParseHistoryBranchInfoRequest{
+			BranchToken: currentVersionHistory.GetBranchToken(),
+		})
+	if err != nil {
+		return err
+	}
+	newHistoryBranchResp, err := r.shard.GetExecutionManager().NewHistoryBranch(ctx,
```

Comment:
nit: Can we add a comment that we need to do this to sanitize the branch token from a different cluster?